gammapy/spectrum/models.py

```diff
@@ -167,6 +167,26 @@ def to_sherpa(self, name='default'):
         """
         raise NotImplementedError('{}'.format(self.__class__.__name__))
 
+    def spectral_index(self, energy):
+        """
+        Compute spectral index at given energy using a local powerlaw
+        approximisation.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            Energy at which to estimate the index
+
+        Returns
+        -------
+        index : float
+            Estimated spectral index.
+        """
+        eps = np.sqrt(np.finfo(float).eps)
```

Comment:
Does such a small `eps` yield numerically stable results?
I think I would have put something like `1e-3`.

But I don't know, if you think or know the spectral index with that small eps gives stable output, OK to keep as-is.